src/Chapter2.hs

```diff
@@ -857,7 +885,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [Int] -> [Int]
+rewind [] = []
+rewind l = last l : rewind (take (length l - 1) l)
```

Comment:
Interesting idea and realisation! 😎 

Though, note, that `last` and `length`are very slow, and this implementation would be quite inefficient.
So you could try to implement it without these functions only using `:`.
For that, you would need to use the `go` pattern with the helper function and accumulator. You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go